lib/teleterm/daemon/daemon.go

```diff
@@ -202,22 +206,28 @@ func (s *Service) RemoveGateway(ctx context.Context, gatewayURI string) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	s.removeGateway(gateway)
+	if err := s.removeGateway(gateway); err != nil {
+		return trace.Wrap(err)
+	}
 
 	return nil
 }
 
 // removeGateway assumes that mu is already held by a public method.
-func (s *Service) removeGateway(gateway *gateway.Gateway) {
-	gateway.Close()
+func (s *Service) removeGateway(gateway *gateway.Gateway) error {
+	if err := gateway.Close(); err != nil {
```

Comment:
Are we ok with not actually removing `gateway` from the gateway list if we get an error on close? Perhaps `removeGateway` should not be responsible for closing a gateway for which we're already holding a pointer for?